packages/plugins/response-cache/src/index.ts

```diff
@@ -102,19 +102,26 @@ export function useResponseCache(options: UseResponseCacheParameter): Plugin {
           session: sessionFactoryForEnvelop,
           buildResponseCacheKey: cacheKeyFactoryForEnvelop,
           shouldCacheResult({ cacheKey, result }) {
-            const shouldCached = options.shouldCacheResult
-              ? options.shouldCacheResult({ cacheKey, result })
-              : !result.errors?.length;
-            if (shouldCached) {
+            let shouldCache: boolean;
+            if (options.shouldCacheResult) {
+              // FIXME: Envelop has wrong type
+              shouldCache = options.shouldCacheResult({ cacheKey, result }) as unknown as boolean;
+            } else {
+              shouldCache = !result.errors?.length;
+              if (!shouldCache) {
+                logger.debug('[useResponseCache] Decided not to cache due to errors');
```

Comment:
This change is breaking because if there is an error with a custom `shouldCache` handler, it ignores the error in favor of `shouldCache`'s result.

Reply:
Ah yes good catch. I'm not sure of what to do about this. Do you think we should keep the old behaviour (never cache if the response contains errors), or do you think it could be interesting to give the freedom to the user to cache responses with errors if it actually wants to ?

Reply:
Perhaps this is a bit too implicit of a behaviour change. If we want to make this changement, we should probably have an additional option here to explicitly allow caching responses with errors.

Reply:
I guess it is a breaking change even to change it from warn to debug. Changing the log message can also be argued to be a breaking change. While I might think that is splitting hairs, its of course up to you how serious you want to go with semver and if it includes logs or not (from case to case).

Anyway, the behaviour in main has now been documented in a spec (in a commit that chronologically happen before even changing any implementation) and that behaviour is:

|  | shouldCacheResult = undefined | shouldCacheResult = false | shouldCacheResult = true |
| ---------|----------|----------|---------|
| errors | skip cache, log warn | skip cache, log warn ❌ | cache, skip log |
| without errors | cache, skip log | skip cache, log warn ❌ | cache, skip log |

This PR changes:

|  | shouldCacheResult = undefined | shouldCacheResult = false | shouldCacheResult = true |
| ---------|----------|----------|---------|
| errors | skip cache, log debug ✅ | skip cache, skip log ⚠️ | cache, skip log |
| without errors | cache, skip log | skip cache, skip log ⚠️ | cache, skip log  |

Do you mean that the breaking change is the ⚠️ cases; i.e. that we no longer log on the user decision to skip the cache?

Are you sure you want to emit a debug log when user decide to skip cache? If so I might need to reword it to be either
- `[useResponseCache] Decided not to cache the response because it contains errors`
- `[useResponseCache] Decided not to cache the response because shouldCacheResult returned false`

Seems a bit silly, I would prefer to just not log it at all. If user wants to they can log themselves, but I dont think this counts as a breaking change at least in my book.

Did I miss something, if so please help me understand 🙏

Reply:
Nevermind, I just missed read the old code. I though it was always skipping on error, whatever `shouldCache` returns. But that's not the case, so there is no breaking changes here, you are right. I think we can merge this :-)

Reply:
Shall we approve then?